drivers/console/websocket_console.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <zephyr/kernel.h>
```

Comment:
Since the zephyr console is a singleton now, I think we might actually need to have DT bindings for this, if we want it to look and feel like a console and get all of the right hooks. But that implies that it would also need to be active immediately on init.

A slightly alternative approach would be to dynamically set up and tear down a zephyr shell per websocket.

Currently, it still seems like the main UART device gets the console / shell.

I could be totally wrong about that though, since I still have not completely gotten the web page to load. Likely I need some iptables rules to forward a port from the physical interface to the tap interface on my build machine.

Reply:
Ok, thanks for the info. I wonder should I remove the actual console support if it is not working as expected and just leave the shell support, wdyt?

Reply:
It might be as easy as defining a Devicetree binding, actually, so could be an easy win.

I enabled the RTT console today, which (in theory) is not very different from the ws console, and that was about as easy to enable.

Reply:
> It might be as easy as defining a Devicetree binding, actually, so could be an easy win.

Ok, I will investigate this.